app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+        channel.setShowBadge(false)
+
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    private fun setSoundVibrationLight(account: AccountEntity, builder: NotificationCompat.Builder) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            return // Do nothing on Android O or newer, the system uses only the channel settings
+        }
+
+        builder.setDefaults(0)
+
+        if (account.notificationSound) {
+            builder.setSound(Settings.System.DEFAULT_NOTIFICATION_URI)
+        }
+
+        if (account.notificationVibration) {
+            builder.setVibrate(longArrayOf(500, 500))
+        }
+
+        if (account.notificationLight) {
+            builder.setLights(-0xd46f27, 300, 1000)
+        }
+    }
+
+    private fun joinNames(notifications1: List<StatusBarNotification>, notifications2: List<Notification>): String? {
+        val names = java.util.ArrayList<String>(notifications1.size + notifications2.size)
+
+        for (notification in notifications1) {
+            val author = notification.notification.extras.getString(EXTRA_ACCOUNT_NAME) ?: continue
+            names.add(author)
+        }
+
+        for (noti in notifications2) {
+            names.add(noti.account.name)
+        }
+
+        if (names.size > 3) {
+            val length = names.size
+            return context.getString(
+                R.string.notification_summary_large,
+                names[length - 1].unicodeWrap(),
+                names[length - 2].unicodeWrap(),
+                names[length - 3].unicodeWrap(),
+                length - 3
+            )
+        } else if (names.size == 3) {
+            return context.getString(
+                R.string.notification_summary_medium,
+                names[2].unicodeWrap(),
+                names[1].unicodeWrap(),
+                names[0].unicodeWrap()
+            )
+        } else if (names.size == 2) {
+            return context.getString(
+                R.string.notification_summary_small,
+                names[1].unicodeWrap(),
+                names[0].unicodeWrap()
+            )
+        }
+
+        return null
+    }
+
+    private fun getStatusReplyIntent(body: Notification, account: AccountEntity, requestCode: Int): PendingIntent {
+        val status = checkNotNull(body.status)
+
+        val inReplyToId = status.id
+        val actionableStatus = status.actionableStatus
+        val replyVisibility = actionableStatus.visibility
+        val contentWarning = actionableStatus.spoilerText
+        val mentions = actionableStatus.mentions
+        var mentionedUsernames: MutableList<String?> = java.util.ArrayList()
+        mentionedUsernames.add(actionableStatus.account.username)
+        for (mention in mentions) {
+            mentionedUsernames.add(mention.username)
+        }
+        mentionedUsernames.removeAll(setOf(account.username))
+        mentionedUsernames = java.util.ArrayList(LinkedHashSet(mentionedUsernames))
```

Comment:
Kotlin has nice list and set builders for cases like this

```suggestion
        val mentionedUsernames = buildSet { 
            add(actionableStatus.account.username)
            for (mention in mentions) {
                add(mention.username)
            }
            remove(account.username)
        }
```